control-plane/subcommand/gateway-resources/command.go

```diff
@@ -331,37 +359,104 @@ func (c *Command) loadConfig() error {
 	return nil
 }
 
+// loadGatewayConfigs reads and loads the configs from `/consul/config/config.yaml`, if this file does not exist nothing is done
+func (c *Command) loadGatewayConfigs() error {
+	file, err := os.Open(c.flagGatewayConfigLocation)
+	if err != nil {
+		if os.IsNotExist(err) {
+			c.UI.Warn(fmt.Sprintf("gateway configuration file not found, skipping gateway configuration, filename: %s", c.flagGatewayConfigLocation))
+			return nil
+		}
+		c.UI.Error(fmt.Sprintf("Error opening gateway configuration file %s: %s", c.flagGatewayConfigLocation, err))
+		return err
+	}
+
+	config, err := io.ReadAll(file)
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error reading gateway configuration file %s: %s", c.flagGatewayConfigLocation, err))
+		return err
+	}
+
+	err = k8syaml.Unmarshal(config, &c.gatewayConfig)
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error decoding gateway config file: %s", err))
+		return err
+	}
+
+	return nil
+}
+
+// createV2GatewayClassAndClassConfigs utilizes the configuration loaded from the gateway config file to
+// create the GatewayClassConfig and GatewayClass for the gateway
```

Comment:
To please the linter that's currently failing

```suggestion
// create the GatewayClassConfig and GatewayClass for the gateway.
```